src/test/java/uk/gov/ons/registers/cdap/service/CompanyHouseServiceTest.java

```diff
@@ -0,0 +1,138 @@
+
+    private static final String TEST_CH_NUMBER = "11240759";
+
+    private static final String TEST_CH_NAME_COLUMN = "companyname";
+    private static final String TEST_CH_NAME = "ANIMAL MICROCHIPS LTD";
+    private static final String TEST_CH_POSTCODE_COLUMN = "regaddress_postcode";
+    private static final String TEST_CH_POSTCODE = "TA4 3NA";
+    private JsonElement TEST_JSON;
+    private ArrayList<JsonElement> TEST_JSON_ARRAYLIST;
+
+    private ServiceManager serviceManager;
+
+    @Before
+    public void setUp() throws Exception {
+        super.beforeTest();
+
+        Map<String, String> testMap = new HashMap<>();
+        testMap.put(TEST_CH_NAME_COLUMN, TEST_CH_NAME);
+        testMap.put(TEST_CH_POSTCODE_COLUMN, TEST_CH_POSTCODE);
+        Gson gson = new Gson();
+        TEST_JSON = gson.toJsonTree(testMap);
+
+        TEST_JSON_ARRAYLIST = new ArrayList<>();
+        TEST_JSON_ARRAYLIST.add(TEST_JSON);
+
+        // Deploy the Sic07 application
+        ApplicationManager appManager = deployApplication(Sic07.class);
+
+        // Get the CompanyHouse dataset
+        DataSetManager<Table> datasetManager = getDataset(Sic07.CH_DATASET_NAME);
+        Table sicCodes = datasetManager.get();
+
+        // Add a Business Number, name and PostCode
+        Put put = new Put(TEST_CH_NUMBER);
+        put.add(TEST_CH_NAME_COLUMN, TEST_CH_NAME);
+        put.add(TEST_CH_POSTCODE_COLUMN, TEST_CH_POSTCODE);
+        sicCodes.put(put);
+
+        // Commit our new row to the dataset
+        datasetManager.flush();
+
+        // Start Sic07Service service
+        serviceManager = appManager.getServiceManager(CompanyHouseService.SERVICE_NAME).start();
+
+        // Wait service startup
+        serviceManager.waitForStatus(true);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        super.afterTest();
+    }
+
+    /**
+     * Business Number Search Test
+     * Returns The Business Information as a JSON object from an entered Company Number
+     */
+
+    @Test
+    public void testChNumberFound() throws Exception {
+        URL url = new URL(serviceManager.getServiceURL(), "CH/bussinesnumber/11240759");
+        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+        assertThat(connection.getResponseCode(), is(HttpURLConnection.HTTP_OK));
+
+        String response;
+        try {
+
+            response = new String(ByteStreams.toByteArray(connection.getInputStream()), Charsets.UTF_8);
+        } finally {
+            connection.disconnect();
+        }
+        assertThat(response, is(TEST_JSON.toString()));
+    }
+
+    @Test
+    public void testChNumberNotFound() throws Exception {
+        URL url = new URL(serviceManager.getServiceURL(), "CH/bussinesnumber/000000");
+        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+        Assert.assertEquals(HttpURLConnection.HTTP_NOT_FOUND, connection.getResponseCode());
```

Comment:
Do you want to use the hamcrest matchers everywhere?

Reply:
Ah, I may have missed a few